### DISCUSSION TITLE
Was RegisteredForRemoteNotifications Removed?

### DISCUSSION THREAD

**Discussion by ElliottBrand** [dotnet/maui#4803] [Q&A]
After upgrading Visual Studio 2022 to Version 17.2.0 Preview 1.0, I can no longer use/override ```RegisteredForRemoteNotifications``` to set my apns token for iOS. If removed, what replaced it?

Prior to this update, I was doing something like:

```csharp
public override void RegisteredForRemoteNotifications(UIApplication application, NSData deviceToken)
{
    var token = deviceToken.ToHexString();
    if (!string.IsNullOrEmpty(token))
        Preferences.Set("apnsToken", token);
}
```

**Comment by Thomr77** [comment]
Hi @ElliottBrand - I wanted to find out if you had any luck with solving this issue?

We have been looking at this too for the last months as background (remote) notification on iOS is key for our product and without it we are stuck at Maui Preview 12 after which these methods were removed. To get around this, we tried to define them ourselves:

`[Export("application:didRegisterForRemoteNotificationsWithDeviceToken:")]`
`public void RegisteredForRemoteNotifications(UIApplication application, NSData deviceToken) {  }`

`[Export("application:didFailToRegisterForRemoteNotificationsWithError:")]`
`public void FailedToRegisterForRemoteNotifications(UIApplication application, NSError error) {  }`

`[Export("application:didReceiveRemoteNotification:fetchCompletionHandler:")]`
`public void DidReceiveRemoteNotification(UIApplication application, NSDictionary userInfo, [BlockProxy(typeof(Trampolines.NIDActionArity1V21))] Action<UIBackgroundFetchResult> completionHandler) {  }`

The first two work fine but DidReceiveRemoteNotification is pretty much impossible to do on your own.

In any case, this seems to be the wrong way to go - do you have any ideas how Maui supports this now?

Please let us know - Thanks!

**Reply by ElliottBrand** [reply]
Sorry, I haven't got it working again and haven't really had time to look into it more. I also implemented the ```application:didRegisterForRemoteNotificationsWithDeviceToken:``` approach, and that does run, but my device isn't displaying new notifications anymore (when the app is in the background).

**Reply by Thomr77** [reply]
@ElliottBrand - Hey check out #6259 - there is a temporary workaround and future fix for this!

Thanks!

**Reply by ElliottBrand** [reply]
Thanks for the follow up!
